egs/wsj/s5/steps/nnet3/get_egs_targets.sh

```diff
@@ -290,16 +330,21 @@ fi
 case $target_type in
   "dense")
     get_egs_program="nnet3-get-egs-dense-targets --num-targets=$num_targets"
-
-    targets="ark:utils/filter_scp.pl --exclude $dir/valid_uttlist $targets_scp_split | copy-feats scp:- ark:- |"
-    valid_targets="ark:utils/filter_scp.pl $dir/valid_uttlist $targets_scp | copy-feats scp:- ark:- |"
-    train_subset_targets="ark:utils/filter_scp.pl $dir/train_subset_uttlist $targets_scp | copy-feats scp:- ark:- |"
+    if [ -z "$scp2ark" ]; then
+      scp2ark="copy-feats scp:- ark:- |"
+    fi
+    targets="ark,s,cs:utils/filter_scp.pl --exclude $dir/valid_uttlist $targets_scp_split | $scp2ark"
```

Comment:
I think maybe the reason you are using the scp2ark thing here is that you didn't realize that you can have scp files as pipes.  For example you could set:
targets="scp:utils/filter.pl --exclude $dir/valid_uttlist $targets_scp_split|"